objects/src/block/block_number.rs

```diff
@@ -0,0 +1,111 @@
+use core::{fmt, ops::Add};
+
+use super::Felt;
+use crate::utils::serde::{
+    ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable,
+};
+
+// BLOCK NUMBER
+// ================================================================================================
+
+/// A convenience wrapper around a `u32` representing the number of a block.
+///
+/// Each block has a unique number and block numbers increase monotonically by `1`.
+#[derive(Debug, Default, Eq, PartialEq, Copy, Clone, PartialOrd, Ord, Hash)]
+pub struct BlockNumber(u32);
+
+impl BlockNumber {
+    /// The length of an epoch expressed as a power of two. `2^(EPOCH_LENGTH_EXPONENT)` is the
+    /// number of blocks in an epoch.
+    ///
+    /// The epoch of a block can be obtained by shifting the block number to the right by this
+    /// exponent.
+    pub const EPOCH_LENGTH_EXPONENT: u8 = 16;
+
+    /// Returns the previous block number
+    pub fn parent(&self) -> u32 {
+        self.checked_sub(1).expect("Cannot go below Genesis block!").0
+    }
```

Comment:
This should return `Option<BlockNumber>`. If you call `BlockNumber::from(0_u32).parent()` this would panic which is not what we want. The caller of `parent()` needs to deal with the possibility of calling this on block number `0`.